plutus-core/plutus-ir/src/PlutusIR/Transform/CommuteFnWithConst.hs

```diff
@@ -0,0 +1,130 @@
+    Term tyname name uni fun a -> Term tyname name uni fun a
+commuteFnWithConst = case eqT @fun @DefaultFun of
+    Just Refl -> commuteDefaultFun
+    Nothing   -> id
+
+-- | Returns whether a `DefaultFun` is commutative with `Constant`'s as arguments. Not using
+-- catchall to make sure that this function catches newly added `DefaultFun`.
+isCommutativeWithConstant :: DefaultFun -> Bool
+isCommutativeWithConstant = \case
+  AddInteger                      -> False
+  SubtractInteger                 -> False
+  MultiplyInteger                 -> True
+  DivideInteger                   -> False
+  QuotientInteger                 -> False
+  RemainderInteger                -> False
+  ModInteger                      -> False
+  EqualsInteger                   -> True
+  LessThanInteger                 -> False
+  LessThanEqualsInteger           -> False
+    -- Bytestrings
+  AppendByteString                -> False
+  ConsByteString                  -> False
+  SliceByteString                 -> False
+  LengthOfByteString              -> False
+  IndexByteString                 -> False
+  EqualsByteString                -> True
+  LessThanByteString              -> False
+  LessThanEqualsByteString        -> False
+    -- Cryptography and hashes
+  Sha2_256                        -> False
+  Sha3_256                        -> False
+  Blake2b_256                     -> False
+  VerifyEd25519Signature          -> False
+  VerifyEcdsaSecp256k1Signature   -> False
+  VerifySchnorrSecp256k1Signature -> False
+    -- Strings
+  AppendString                    -> False
+  EqualsString                    -> True
+  EncodeUtf8                      -> False
+  DecodeUtf8                      -> False
+    -- Bool
+  IfThenElse                      -> False
+    -- Unit
+  ChooseUnit                      -> False
+    -- Tracing
+  Trace                           -> False
+    -- Pairs
+  FstPair                         -> False
+  SndPair                         -> False
+    -- Lists
+  ChooseList                      -> False
+  MkCons                          -> False
+  HeadList                        -> False
+  TailList                        -> False
+  NullList                        -> False
+    -- Data
+  ChooseData                      -> False
+  ConstrData                      -> False
+  MapData                         -> False
+  ListData                        -> False
+  IData                           -> False
+  BData                           -> False
+  UnConstrData                    -> False
+  UnMapData                       -> False
+  UnListData                      -> False
+  UnIData                         -> False
+  UnBData                         -> False
+  EqualsData                      -> False -- doesn't take constant
+  SerialiseData                   -> False
+  MkPairData                      -> False
+  MkNilData                       -> False
+  MkNilPairData                   -> False
```

Comment:
I'm unsure, but I think given we're listing some exceptions, this could reasonably have a `_ -> False` case.

Reply:
What I'm trying to do is to force us to add a case here whenever we add a new `DefaultFun`. Otherwise, any new builtins will just be false, because we're unlikely to remember this unless the compiler reminds us. Of course, not remembering is not the end of the world because we just won't commute, but I think it makes sense to add any upcoming commutative functions in here.